src/infrastructure/repositories/ViemChainMetricsRepository.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  createPublicClient,
+  http,
+  PublicClient
+} from "viem"
+// @ts-ignore
+import { hemiSepolia, hemi } from "hemi-viem"
+// @ts-ignore
+import { Hash } from '../../domain/valueObjects/Hash'
+import { ChainMetricsRepository } from '../../domain/repositories/ChainMetricsRepository'
+import { ChainMetrics } from '../../domain/entities/ChainMetrics'
+import { BlockNumber } from '../../domain/valueObjects/BlockNumber'
+import { Metric } from '../../domain/valueObjects/Metric'
+
+export class ViemChainMetricsRepository implements ChainMetricsRepository {
+  private client: PublicClient
+  private blockExplorerUrl: string
+
+  constructor() {
+    const chain = !!process.env['TESTNET'] ? hemiSepolia : hemi
+    
+    // @ts-expect-error
+    this.client = createPublicClient({ chain, transport: http() })
+    this.blockExplorerUrl = process.env['BLOCK_EXPLORER_URL'] || ''
+  }
+
+  async collect(): Promise<ChainMetrics> {
+    const blockNumber = await this.client.getBlockNumber()
+    
+    const response = await fetch(`${this.blockExplorerUrl}/api/v2/stats`)
+    const stats = await response.json()
```

Comment:
if you want to omit this line (?) you can use our super duper [fetch-plus-plus](https://www.npmjs.com/package/fetch-plus-plus), which runs `.json()` for you 😄 

If you don't feel like it (if this is the only `fetch` call, it may seem overkill to just install a dependency for this), another way to save one line and discard from the scope of the whole function the `response` variable (which you only need to call `.json`) is to combine `await `with `.then()`.

Combining this suggestion with the [above comment](https://github.com/hemilabs/HemiConnector/pull/3/files#r1899769835), you could rewrite everything as

```ts
const [blockNumber, response] = await Promise.all([
  this.client.getBlockNumber(), 
  fetch(`${this.blockExplorerUrl}/api/v2/stats`).then(response => response.json())
])
```

This way `response` is now only local to the fetch promise (And both calls are in parallel).


These are different suggestions, choose your own path here! 😄